tests/Doctrine/Tests/DBAL/Functional/WriteTest.php

```diff
@@ -255,6 +280,13 @@ public function testEmptyIdentityInsert() : void
             );
         }
 
+        if ($platform instanceof OraclePlatform && $this->connection->getWrappedConnection() instanceof PDOConnection) {
+            $this->markTestSkipped(
+                'Oracle supports emulated identity columns through sequences, ' .
```

Comment:
It doesn't look like that. It throws an exception if `lastInsertId()` is called without a name. Neither of Oracle drivers support this API. It should be probably detected based on `AbstractOracleDriver`.

Reply:
You may be wrong here. [The PDO OCI test failed](https://app.continuousphp.com/git-hub/doctrine/dbal/build/2454fff4-ac55-4feb-8238-bf2b19615841/logs/50ff3103-64d5-4395-a9c8-7194766b2d60), even though the stack trace pointed to:

https://github.com/doctrine/dbal/blob/ea93b5542d15b14db596a7fc758ffe4fef04c3bb/tests/Doctrine/Tests/DBAL/Functional/WriteTest.php#L189

Then:

https://github.com/doctrine/dbal/blob/ea93b5542d15b14db596a7fc758ffe4fef04c3bb/lib/Doctrine/DBAL/Driver/PDOConnection.php#L116

Where you can see that `lastInsertId()` is called with a name.

My suspicion is in the exception message: Oracle supports sequences, but PDO OCI doesn't support them through `lastInsertId()`.

Reply:
In any event, the error message "Oracle supports emulated identity columns through sequences" is misleading. The last inserted ID is scoped to a connection and has one value for the entire connection while sequences are persisted in the database and there may be more than one of them. It's hard to imagine how a driver could emulate identity columns through sequences.

Reply:
I just followed the [already established convention](https://github.com/doctrine/dbal/blob/299ef4f06645b747fbef7876f87516012eb990ba/lib/Doctrine/DBAL/Platforms/AbstractPlatform.php#L2689). What do you suggest exactly?

Reply:
I don't understand what this method means. Besides the boolean value returned by these methods, I don't see any emulation happening on the platform level. This method is not used anywhere except in tests.

I suggest changing the condition to:
```php
if ($driver instanceof AbstractOracleDriver) {
    $this->markTestSkipped('The driver does not support lastInsertId()');
}
```
There's nothing about `lastInsertId()` on the platform level, IMO.

Reply:
TBH I only tried to accommodate existing code as much as possible. After more review though, you're right that `usesSequenceEmulatedIdentityColumns()` is only used in tests. Do you want me to attempt to remove it altogether, in a separate PR?

Then `testEmptyIdentityInsertEmulated()` would be gone here.

Reply:
I'd rather have it removed in a separate PR. We can merge it first and then rebase this one.

Reply:
No sooner said than done: #3769

Reply:
Thank you for filing the ticket. I'm going to take some time and learn the codebase better. I'll get back to you as soon as I have a solid understanding of how exactly I'd like us to proceed.

Reply:
> Oracle supports emulated identity columns through sequences

Okay. According to [this respectful source](https://en.wikipedia.org/wiki/Identity_column) an identity column is the one that is made up of values generated by the database. So [Oracle does emulate identity columns through sequences](https://github.com/doctrine/dbal/blob/436f2f6c943937abe8544505630e76be4c468ae1/lib/Doctrine/DBAL/Platforms/OraclePlatform.php#L395) which is a platform feature in the DBAL terminology. 

Although I still fail to see how this is relevant to the `lastInsertId()` API which is a driver feature. The exception message is still misleading to me.

Reply:
What do you suggest?

Reply:
The same as in https://github.com/doctrine/dbal/pull/3759#discussion_r352818386.

Reply:
Done in 7ec283d.